### PR TITLE
feat(api-gateway): support non api gateway

### PR DESCRIPTION
**Pull request by argadityars**
Closes CLD-973

### CODE REVIEW

File: widget/src/utils/env.ts [hunk 1]
```diff
@@ -8,21 +8,21 @@
 
 export const isDevMode = process.env.NODE_ENV === "development";
 
-export const redirectURI =
-  process.env.CLIENT_REDIRECT_URI || (isDevMode && "https://demo.accelbyte.io") || "";
+export const redirectURI = process.env.CLIENT_REDIRECT_URI || (isDevMode && "https://versusevil.com") || "";
```
Comment (klepon) [inline widget/src/utils/env.ts]: can we revert this into demo.accelbyte.io instead?
also for other changes below

**Review by klepon — CHANGES_REQUESTED** [review]
got error

**Unhandled Rejection (TypeError): Cannot read property 'avatarUrl' of undefined**

UserProfileButton.render
src/components/Auth/UserProfileButton.tsx:83
```
  80 | >
  81 |   <Avatar
  82 |     isLoading={isFetchingProfile}
> 83 |     imageUrl={profile!.avatarUrl}
  84 | ^   className={classNames(
  85 |       "ab-wpr-avatar",
  86 |       {"large": screenWidth >= breakPoint.medium.max},
```

File: widget/src/utils/env.ts (hunk 1 above)
Comment (argadityars) [reply widget/src/utils/env.ts]: ah my bad, will update it to demo

File: README.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ To build this plugin you need to set up following environments variable:
 - `CLIENT_REDIRECT_URI` - URI to redirect back from AccelByte Login Website to WordPress website
 - `APP_SECURITY_CLIENT_ID` - Client ID from IAM for WordPress application in browser
 - `JUSTICE_PUBLISHER_NAMESPACE` - Default namespace
-- `JUSTICE_APIGATEWAY_BASE_URL` - URL of your AccelByte APIGateway
+- `JUSTICE_BASE_URL` - URL of your AccelByte APIGateway
```
Comment (THZs) [inline README.md]: We don't use API Gateway anymore, how if we change the description to "Base URL of Accelbyte services"?

**Comment by argadityars** [comment]
> got error
> 
> **Unhandled Rejection (TypeError): Cannot read property 'avatarUrl' of undefined**
> 
> UserProfileButton.render
> src/components/Auth/UserProfileButton.tsx:83
> 
> ```
>   80 | >
>   81 |   <Avatar
>   82 |     isLoading={isFetchingProfile}
> > 83 |     imageUrl={profile!.avatarUrl}
>   84 | ^   className={classNames(
>   85 |       "ab-wpr-avatar",
>   86 |       {"large": screenWidth >= breakPoint.medium.max},
> ```

I didn't make any changes into this file, and actually I can build the project without getting this error. can you tell me how to reproduce this?

**Comment by klepon** [comment]
> > got error
> > **Unhandled Rejection (TypeError): Cannot read property 'avatarUrl' of undefined**
> > UserProfileButton.render
> > src/components/Auth/UserProfileButton.tsx:83
> > ```
> >   80 | >
> >   81 |   <Avatar
> >   82 |     isLoading={isFetchingProfile}
> > > 83 |     imageUrl={profile!.avatarUrl}
> >   84 | ^   className={classNames(
> >   85 |       "ab-wpr-avatar",
> >   86 |       {"large": screenWidth >= breakPoint.medium.max},
> > ```
> 
> I didn't make any changes into this file, and actually I can build the project without getting this error. can you tell me how to reproduce this?

i see this from current master branch, is nice if this can be fixed as well, it fixed using `imageUrl={profile ? profile!.avatarUrl : ""}`
the error occur on browser when run `yarn start` or after build when it use on WP the js script error causing it not working

btw it happen on SBZ project

**Comment by argadityars** [comment]
> > > got error
> > > **Unhandled Rejection (TypeError): Cannot read property 'avatarUrl' of undefined**
> > > UserProfileButton.render
> > > src/components/Auth/UserProfileButton.tsx:83
> > > ```
> > >   80 | >
> > >   81 |   <Avatar
> > >   82 |     isLoading={isFetchingProfile}
> > > > 83 |     imageUrl={profile!.avatarUrl}
> > >   84 | ^   className={classNames(
> > >   85 |       "ab-wpr-avatar",
> > >   86 |       {"large": screenWidth >= breakPoint.medium.max},
> > > ```
> > 
> > 
> > I didn't make any changes into this file, and actually I can build the project without getting this error. can you tell me how to reproduce this?
> 
> i see this from current master branch, is nice if this can be fixed as well, it fixed using `imageUrl={profile ? profile!.avatarUrl : ""}`
> the error occur on browser when run `yarn start` or after build when it use on WP the js script error causing it not working
> 
> btw it happen on SBZ project

okay, it's updated now

**Review by klepon — APPROVED** [review]

**Comment by THZs** [comment]
> got error
> 
> **Unhandled Rejection (TypeError): Cannot read property 'avatarUrl' of undefined**
> 
> UserProfileButton.render
> src/components/Auth/UserProfileButton.tsx:83
> 
> ```
>   80 | >
>   81 |   <Avatar
>   82 |     isLoading={isFetchingProfile}
> > 83 |     imageUrl={profile!.avatarUrl}
>   84 | ^   className={classNames(
>   85 |       "ab-wpr-avatar",
>   86 |       {"large": screenWidth >= breakPoint.medium.max},
> ```

What @klepon mentioned here is correct, and thank you for raising this up. I think I missed this one during writing this.
It should be using [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining) instead of [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator). Can you please help to update it also @argaditya-ab ?
I think by changing the `!` mark to `?` would be enough. Thank you!

**Review by THZs — APPROVED** [review]

**Pull request merged**